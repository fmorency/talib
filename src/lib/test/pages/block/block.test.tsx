import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Block from "lib/pages/block/index";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("Home", () => {
  const component = renderer.create(<Block />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
