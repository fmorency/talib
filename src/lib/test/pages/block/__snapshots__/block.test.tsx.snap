// Vitest Snapshot v1

exports[`Home 1`] = `
[
  <div
    className="css-19bewft"
  >
    <div
      className="css-0"
    >
      <div
        className="css-i7slzv"
      >
        <h3
          className="chakra-heading css-13jwqxv"
        >
          BLOCK
        </h3>
        <p
          className="chakra-text css-z1c0pb"
        >
          #123456
        </p>
      </div>
      <div
        className="css-1gfvzb4"
      >
        <div
          className="css-tmma79"
        >
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                2 hs 1 min ago
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              />
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              TIME
            </p>
          </div>
        </div>
        <div
          className="css-tmma79"
        >
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-15wnhow"
              >
                545,774,135,429.02
              </p>
              <p
                className="chakra-text css-1g027ah"
              >
                usd
              </p>
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              NUMBER OF TRANSACTIONS
            </p>
          </div>
        </div>
        <div
          className="css-tmma79"
        >
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                545,774,135,429.02
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              />
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              SIZE DATA
            </p>
          </div>
        </div>
        <div
          className="css-tmma79"
        >
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                12345678
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              />
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              NODES
            </p>
          </div>
        </div>
        <div
          className="css-tmma79"
        >
          <div
            className="css-ipfnbh"
          >
            <div
              className="chakra-stack css-12t3lkn"
            >
              <p
                className="chakra-text css-1kf9ore"
              >
                26547789
              </p>
              <p
                className="chakra-text css-1yl1k0u"
              />
            </div>
            <p
              className="chakra-text css-jf0soh"
            >
              VALIDATORS
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-za6arg"
  >
    <div
      className="css-imrbmu"
    >
      <h3
        className="chakra-heading css-1i80fgp"
      >
        History
      </h3>
      <div
        className="css-1j3ysxt"
      >
        <div
          className="css-5pj2v0"
        >
          <div
            className="css-u35zva"
          >
            <table
              className="chakra-table css-qegi10"
              role="table"
            >
              <thead
                className="css-0"
              >
                <tr
                  className="css-0"
                  role="row"
                >
                  <th
                    className="css-0"
                    colSpan={1}
                    role="columnheader"
                  >
                    TSX / HASH
                  </th>
                  <th
                    className="css-0"
                    colSpan={1}
                    role="columnheader"
                  >
                    TIME
                  </th>
                  <th
                    className="css-0"
                    colSpan={1}
                    role="columnheader"
                  >
                    TO
                  </th>
                  <th
                    className="css-0"
                    colSpan={1}
                    role="columnheader"
                  >
                    FROM
                  </th>
                  <th
                    className="css-0"
                    colSpan={1}
                    role="columnheader"
                  >
                    TYPE
                  </th>
                </tr>
              </thead>
              <tbody
                className="css-0"
                role="rowgroup"
              >
                <tr
                  className="css-0"
                  role="row"
                >
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/block/457676765"
                    >
                      #457676765
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      2hs 36min ago
                    </span>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646667"
                    >
                      0x457676765
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646667"
                    >
                      0x457676765
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      Buy
                    </span>
                  </td>
                </tr>
                <tr
                  className="css-0"
                  role="row"
                >
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/block/457676766"
                    >
                      #457676766
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      4hs 15mins ago
                    </span>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646668"
                    >
                      0x457676766
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646668"
                    >
                      0x457676766
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      Sell
                    </span>
                  </td>
                </tr>
                <tr
                  className="css-0"
                  role="row"
                >
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/block/457676767"
                    >
                      #457676767
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      4hs 54mins ago
                    </span>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646669"
                    >
                      0x457676767
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646669"
                    >
                      0x457676767
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      Sign
                    </span>
                  </td>
                </tr>
                <tr
                  className="css-0"
                  role="row"
                >
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/block/457676768"
                    >
                      #457676768
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      6hs 46mins ago
                    </span>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646670"
                    >
                      0x457676768
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646670"
                    >
                      0x457676768
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      Execute
                    </span>
                  </td>
                </tr>
                <tr
                  className="css-0"
                  role="row"
                >
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/block/457676769"
                    >
                      #457676769
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      4hs 16mins ago
                    </span>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646671"
                    >
                      0x457676769
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <a
                      className="chakra-link css-zamdaj"
                      href="/address/5435364646671"
                    >
                      0x457676769
                    </a>
                  </td>
                  <td
                    className="css-0"
                    role="cell"
                  >
                    <span>
                      Buy
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            className="css-37qczb"
          >
            <div
              className="chakra-stack css-rlwq96"
            >
              <div
                className="chakra-stack css-1ss2m3q"
              >
                <button
                  className="chakra-button css-dsj5zd"
                  type="button"
                >
                  &lt;&lt;
                </button>
                <button
                  className="chakra-button css-gi0s1o"
                  disabled={true}
                  type="button"
                >
                  1
                </button>
                <button
                  className="chakra-button css-59095b"
                  onClick={[Function]}
                  type="button"
                >
                  2
                </button>
                <button
                  className="chakra-button css-59095b"
                  onClick={[Function]}
                  type="button"
                >
                  &gt;&gt;
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
